test/helpers/client.js

```diff
@@ -44,17 +44,22 @@ class Client {
         this._respondWithCallback(JSON.parse(data));
     }
 
+    _setupMessageEvents() {
+        this.ws.on('open', this._onOpen.bind(this));
+        this.ws.on('close', this._onClose.bind(this));
+        this.ws.on('message', this._onMessage.bind(this));
+    }
+
     _checkError(error) {
         // Assumes that an instance of the server is not running, so lets start one.
-        if (error.errno === "ECONNREFUSED") {
+        if (error.errno === 'ECONNREFUSED') {
             if (!this.child) {
-                this.child = fork('main.js', ['grunt', '--debug', 'silly'])
+                this.child = fork('main.js', ['grunt', '--debug', 'silly']);
```

Comment:
I tried to run it as a spawn process, but it didn't work, the socket refused to with connection reset, on socket hangup.